src/components/MapName.js

```diff
@@ -0,0 +1,40 @@
+import React, { useState, useRef, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+
+const MapName = ({ }) => {
+    const [editing, setEditing] = useState(false);
+
+    const dispatch = useDispatch();
+    const mapName = useSelector((state) => state.map.name);
+
+    const ref = useRef();
+
+    useEffect(() => {
+        const checkIfClickedOutside = e => {
```

Comment:
React doesn't want to give a <p> an onfocusout event.

The enter key gives a line break at the moment, I see you've cleared that up in the drawing PR. I think it'd be easier to integrate an onKeyDown event after that's been merged and I'll follow the styles there

Reply:
All the different title bar issues are confusing me, so going to close this PR.
I'll include the changes mentioned here in #172 since I'm working with some of this code anywyay.